Sources/Testing/Parameterization/Test.Case.ID.swift

```diff
@@ -0,0 +1,44 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+//
+
+@_spi(ExperimentalParameterizedTesting)
+extension Test.Case: Identifiable {
+  /// The ID of a test case.
+  ///
+  /// Instances of this type are considered unique within the scope of a given
+  /// parameterized test function. They are not necessarily unique across two
+  /// different ``Test`` instances.
+  public struct ID: Sendable, Equatable, Hashable {
+    /// The IDs of the arguments of this instance's associated ``Test/Case``, in
+    /// the order they appear in ``Test/Case/arguments``.
+    public var argumentIDs: [String]
```

Comment:
Does this behave correctly for monomorphic test functions (where the set of arguments is the empty array and the test has a single case)?

Reply:
Yeah, it's represented as a "trivial" `Test.Case.ID` containing an empty `argumentIDs` array.